# Data Types and File Formats

GDC provides access to data submitted from new programs, data generated by GDC through GDC alignment and higher-level data generation pipelines, and data imported from existing programs.

## Submitted Data

GDC currently accepts DNA, RNA and miRNA sequencing data in both [FASTQ](http://maq.sourceforge.net/fastq.shtml) and [BAM](https://samtools.github.io/hts-specs/SAMv1.pdf) formats. Sequencing data is submitted with accompanying metadata in either simple tab-separated (TSV) format, or [JavaScript Object Notation (JSON)](http://http://www.json.org//) format. Clinical and Biospecimen data can be submitted in either TSV format, JSON format, or as XML that is validated with respect to the latest version (currently 2.6) of [NCI Biospecimen Core Resource XML Schema documents](https://tcga-data.nci.nih.gov/docs/xsd/BCR/tcga.nci/bcr/xml/).  These data may also be provided to GDC via the GDC Data Submission Portal.

<div class="read-more-button doc">

[<span class="doc-icon"></span><span class="doc-name">Submitted Data Types and File Formats</span>](/node/8396/)

</div>

## Generated Data

For all submitted sequence data, including aligned BAM files, GDC generates new alignments in BAM format using the latest human reference genome [GRCh38](http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/human/) with [standard pipelines](/node/8374/). Original submitted data is retained. Using these standard alignments, GDC generates higher-level derived data, including normal and tumor variant and mutation calls in [VCF](https://wiki.nci.nih.gov/display/TCGA/TCGA+Variant+Call+Format+%28VCF%29+Specification) and [MAF](https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+%28MAF%29+Specification+-+v2.4) formats, and digital gene expression, miRNA expression, exon-junction quantification data in TSV formats.

<div class="read-more-button doc">

[<span class="doc-icon"></span><span class="doc-name">Generated Data Types and File Formats</span>](/node/8397/)

</div>

## Imported Data

GDC hosts and distributes previously generated data from [The Cancer Genome Atlas (TCGA)](http://cancergenome.nih.gov/), [Therapeutically Applicable Research to Generate Effective Treatments (TARGET)](https://ocg.cancer.gov/programs/target), and the [Cancer Genome Characterization Initiative (CGCI)](https://ocg.cancer.gov/programs/cgci) programs. Original sequence alignments are stored in BAM format, and derived data files are stored and provided in their original formats.

<div class="read-more-button doc">

[<span class="doc-icon"></span><span class="doc-name">Imported Data Types and File Formats</span>](/node/8395/)

</div>

# Generated Data Types and File Formats

The following table lists data type and data subtype categories used to classify data generated by GDC. **Not all programs, projects, or cases will have data available for all types.**

Generated Data Types and File Formats:

| Data Type | Data Subtype | Format |
| --- | --- | --- |
| Raw Sequencing Data | Aligned Reads | [BAM](http://samtools.github.io/hts-specs/SAMv1.pdf) |
| Simple Nucleotide Variation | Genotypes | TSV |
| Simple Germline Variation | [MAF](https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+%28MAF%29+Specification+-+v2.4), [VCF](https://wiki.nci.nih.gov/display/TCGA/TCGA+Variant+Call+Format+%28VCF%29+Specification) |
| Simple Somatic Mutation |
| Gene Expression | Gene expression Quantification | TSV |
| miRNA Quantification |
| Isoform Expression Quantification |
| Exon Junction Quantification |
| Exon Quantification |
| Structural Rearrangement | Structural Variation | [VCF](https://wiki.nci.nih.gov/display/TCGA/TCGA+Variant+Call+Format+%28VCF%29+Specification), [FASTA](http://www.ncbi.nlm.nih.gov/BLAST/blastcgihelp.shtml) |
| Copy Number Variation | Copy Number Segmentation | TSV |

# Imported Data Types and File Formats

The following table lists major data type and data subtype categories used to classify imported data files available to users through GDC. **Not all programs, projects, or cases will have data available for all types.**

Imported Data Types and File Formats:

| Data Type | Data Subtype | Format |
| --- | --- | --- |
| Raw Sequencing data | Aligned Reads | [BAM](http://samtools.github.io/hts-specs/SAMv1.pdf) |
| Unaligned Reads | [FASTQ](http://maq.sourceforge.net/fastq.shtml) |
| Coverage WIG | [WIGGLE](http://genome.ucsc.edu/goldenpath/help/wiggle.html) |
| <span style="font-size: 13.3333px; line-height: 20px;">Simple Nucleotide Variation</span> | Genotypes | TSV |
| Simple Germline Variation | [MAF](https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+%28MAF%29+Specification+-+v2.4), [VCF](https://wiki.nci.nih.gov/display/TCGA/TCGA+Variant+Call+Format+%28VCF%29+Specification) |
| Simple Somatic Mutation |
| Raw Microarray Data | Raw Intensities | TSV |
| CGH Array QC |
| Intensities Log2Ratio |
| Expression Control |
| Intensities |
| Normalized Intensities |
| Probeset Summary |
| Methylation Array QC Metrics |
| Gene Expression | Gene Expression Quantification | TSV |
| miRNA Quantification |
| Isoform Expression Quantification |
| Exon Junction Quantification |
| Exon Quantification |
| Gene Expression Summary |
| Structural Rearrangement | Structural Variation | [VCF](https://wiki.nci.nih.gov/display/TCGA/TCGA+Variant+Call+Format+%28VCF%29+Specification), [FASTA](http://www.ncbi.nlm.nih.gov/BLAST/blastcgihelp.shtml) |
| DNA Methylation | Bisulfite Sequence Alignment | [BAM](http://samtools.github.io/hts-specs/SAMv1.pdf) |
| Methylation Beta Value | TSV |
| Methylation Percentage |
| Clinical | Clinical Data | XML |
| Biospecimen Data |
| Tissue Slide Image | SVS |
| Diagnostic Image |
| Pathology Report | PDF |
| Copy Number Variation | Copy Number Segmentation | TSV |
| Copy Number Estimate |
| LOH |
| Copy Number QC Metrics |
| Copy Number Variation |
| Normalized Copy Numbers |
| Copy Number Summary |
| Probeset Call |
| Protein Expression | Protein Expression Quantification | TSV |
| Protein Expression Control |
| Other | Microsattelite Instability | FSA |
| ABI Sequence Trace | TR |
| Auxiliary Test | TSV, XML, HTML |

# Submitted Data Types and File Formats

The following table lists data type and data subtype categories for data submitted into GDC and provides the required file formats along with example files.  **Not all programs, projects, or cases will have data available for all types.**

Submitted Data Types and File Formats:

| Data Type | Data Subtype | Format | Data Dictionary | Example File |
| --- | --- | --- | --- | --- |
| Biospecimen | Biospecimen Data | XML, Tab-delimited, JSON |
| Clinical | Clinical Data | XML, Tab-delimited, JSON | [Clinical Data Elements](/node/8216/) |
| Tissue Slides | SVS | N/A |
| Pathology Reports | PDF | N/A |
| Raw Sequencing Data | DNA | [BAM](http://samtools.github.io/hts-specs/SAMv1.pdf), [FASTQ](http://maq.sourceforge.net/fastq.shtml) |
| mRNA | [BAM](http://samtools.github.io/hts-specs/SAMv1.pdf), [FASTQ](http://maq.sourceforge.net/fastq.shtml) |
| miRNA | [BAM](http://samtools.github.io/hts-specs/SAMv1.pdf), [FASTQ](http://maq.sourceforge.net/fastq.shtml) |
| Raw Sequencing Metadata | DNA, mRNA, miRNA | TSV, JSON |
